homeassistant/components/broadlink/climate.py

```diff
@@ -0,0 +1,89 @@
+"""Support for Broadlink climate devices."""
+from typing import Any
+
+from homeassistant.components.climate import (
+    ATTR_TEMPERATURE,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACAction,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PRECISION_HALVES, Platform, UnitOfTemperature
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.restore_state import RestoreEntity
+
+from .const import DOMAIN, DOMAINS_AND_TYPES
+from .device import BroadlinkDevice
+from .entity import BroadlinkEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Broadlink climate entities."""
+    device = hass.data[DOMAIN].devices[config_entry.entry_id]
+
+    if device.api.type in DOMAINS_AND_TYPES[Platform.CLIMATE]:
+        async_add_entities([BroadlinkThermostat(device)])
+
+
+class BroadlinkThermostat(ClimateEntity, BroadlinkEntity, RestoreEntity):
+    """Representation of a Broadlink Hysen climate entity."""
+
+    _attr_has_entity_name = True
+    _attr_hvac_modes = [HVACMode.HEAT, HVACMode.OFF, HVACMode.AUTO]
+    _attr_supported_features = ClimateEntityFeature.TARGET_TEMPERATURE
+    _attr_target_temperature_step = PRECISION_HALVES
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+
+    def __init__(self, device: BroadlinkDevice) -> None:
+        """Initialize the climate entity."""
+        super().__init__(device)
+        self._attr_unique_id = device.unique_id
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        temperature = kwargs[ATTR_TEMPERATURE]
+        await self._device.async_request(self._device.api.set_temp, temperature)
+        self._attr_target_temperature = temperature
+        self.async_write_ha_state()
+
+    @callback
+    def _update_state(self, data: Any) -> None:
```

Comment:
```suggestion
    def _update_state(self, data: dict[str, Any]) -> None:
```

Otherwise your code will raise errors